bug: All formatters fail when injected formatter fails

### Neovim version (nvim -v)

NVIM v0.9.2 Build type: Release LuaJIT 2.1.1694285958

### Operating system/version

Arch Linux 6.5.5-arch1-1

### Add the debug logs

- [X] I have set `log_level = vim.log.levels.DEBUG` and pasted the log contents below.

### Log file

Log file: /home/rileyb/.local/state/nvim/conform.log
            XDG_CONFIG_HOME = "/home/rileyb/.config/nvim/"
          }
          20:59:45[INFO] stylua exited with code 2
          20:59:45[DEBUG] stylua stdout: { "" }
          20:59:45[DEBUG] stylua stderr: { "error: could not format from stdin: failed to format from stdin: error parsing: error occurred while creating ast: unexpected token `>`. (starting from line 1, character 1 and ending on line 1, character 2)", "additional information: leftover token", "" }
          20:59:45[ERROR] Formatter 'stylua' error: error: could not format from stdin: failed to format from stdin: error parsing: error occurred while creating ast: unexpected token `>`. (starting from line 1, character 1 and ending on line 1, character 2)
          additional information: leftover token
          
          20:59:45[ERROR] Formatter 'stylua' error: error: could not format from stdin: failed to format from stdin: error parsing: error occurred while creating ast: unexpected token `>`. (starting from line 1, character 1 and ending on line 1, character 2)
          additional information: leftover token

Formatters for this buffer:
deno_fmt ready (javascript, javascriptreact, typescriptreact, markdown, typescript, jsonc, json)
injected ready (markdown)

Other formatters:
clang_format ready (c, cpp)
prettierd ready (css, html)
stylua ready (luau, lua)


### Describe the bug

When using injected formatters, if an injected formatter fails then the main formatter(s) will not run. For example if an injected code block in a Markdown document cannot be formatted, the entire Markdown document will not be formatted.

### Steps To Reproduce

1. `nvim -u repro.lua`
2. Open a file with injected formatting that will fail
3. Save the file

### Expected Behavior

The Lua code will not be formatted, but the rest of the markdown document will

### Minimal example file

````md
# Hi

   wrongly indented text

> ```lua
> require('hi')
> ```

The above cannot be formatted since it is in a block quote :)
````

### Minimal init.lua

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify('./.repro', ':p')

-- set stdpaths to use .repro
for _, name in ipairs { 'config', 'data', 'state', 'cache' } do
  vim.env[('XDG_%s_HOME'):format(name:upper())] = root .. '/' .. name
end

-- bootstrap lazy
local lazypath = root .. '/plugins/lazy.nvim'
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system {
    'git',
    'clone',
    '--filter=blob:none',
    '--single-branch',
    'https://github.com/folke/lazy.nvim.git',
    lazypath,
  }
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  'folke/tokyonight.nvim',
  {
    'stevearc/conform.nvim',
    config = function()
      require('conform').setup {
        log_level = vim.log.levels.DEBUG,
        format_on_save = {
          timeout_ms = 1000,
          lsp_fallback = false,
          quiet = true,
        },
        formatters_by_ft = {
          lua = { 'stylua' },
          markdown = { 'deno_fmt', 'injected' },
        },
      }
    end,
  },
  -- add any other plugins here
}
require('lazy').setup(plugins, {
  root = root .. '/plugins',
})

vim.cmd.colorscheme('tokyonight')
-- add anything else here
```


### Additional context

_No response_

--- a/lua/conform/formatters/injected.lua
+++ b/lua/conform/formatters/injected.lua
@@ -7,12 +7,18 @@ local function in_range(range, start_lnum, end_lnum)
 end
 
 ---@param lines string[]
+---@param language? string The language of the buffer
 ---@return string?
-local function get_indent(lines)
+local function get_indent(lines, language)
   local indent = nil
+  -- Handle markdown code blocks that are inside blockquotes
+  -- > ```lua
+  -- > local x = 1
+  -- > ```
+  local pattern = language == "markdown" and "^>?%s*" or "^%s*"
   for _, line in ipairs(lines) do
     if line ~= "" then
-      local whitespace = line:match("^%s*")
+      local whitespace = line:match(pattern)
       if whitespace == "" then
         return nil
       elseif not indent or whitespace:len() < indent:len() then
@@ -23,20 +29,33 @@ local function get_indent(lines)
   return indent
 end
 
----@param original_lines string[]
----@param new_lines? string[]
-local function apply_indent(original_lines, new_lines)
-  if not new_lines then
+---Remove leading indentation from lines and return the indentation string
+---@param lines string[]
+---@param language? string The language of the buffer
+---@return string?
+local function remove_indent(lines, language)
+  local indent = get_indent(lines, language)
+  if not indent then
     return
   end
-  local indent = get_indent(original_lines)
-  local already_indented = get_indent(new_lines)
-  if not indent or already_indented then
-    return
-  end
-  for i, line in ipairs(new_lines) do
+  local sub_start = indent:len() + 1
+  for i, line in ipairs(lines) do
     if line ~= "" then
-      new_lines[i] = indent .. line
+      lines[i] = line:sub(sub_start)
+    end
+  end
+  return indent
+end
+
+---@param lines string[]?
+---@param indentation string?
+local function apply_indent(lines, indentation)
+  if not lines or not indentation then
+    return
+  end
+  for i, line in ipairs(lines) do
+    if line ~= "" then
+      lines[i] = indentation .. line
     end
   end
 end
@@ -135,9 +154,10 @@ return {
         local idx = num_format
         log.debug("Injected format %s:%d:%d: %s", lang, start_lnum, end_lnum, formatter_names)
         log.trace("Injected format lines %s", input_lines)
+        local indent = remove_indent(input_lines, buf_lang)
         conform.format_lines(formatter_names, input_lines, format_opts, function(err, new_lines)
           -- Preserve indentation in case the code block is indented
-          apply_indent(input_lines, new_lines)
+          apply_indent(new_lines, indent)
           formatter_cb(err, idx, start_lnum, end_lnum, new_lines)
         end)
       end

